### PR TITLE
Document SQLP supported isolation levels

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 1]
```diff
@@ -24,7 +24,28 @@ The trade-off is that each message processing attempt requires additional round
 
 ## Transaction type
 
-By default the outbox uses the ADO.NET transactions abstracted via `DbTransaction`. This is appropriate for most situations.
+The outbox supports transactions via [ADO.NET Local transactions](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/local-transactions) or [TransactionScopes](https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscope).
+
+### ADO.net
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:29]:
```suggestion
### ADO.NET
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 2]
```diff
@@ -24,7 +24,28 @@ The trade-off is that each message processing attempt requires additional round
 
 ## Transaction type
 
-By default the outbox uses the ADO.NET transactions abstracted via `DbTransaction`. This is appropriate for most situations.
+The outbox supports transactions via [ADO.NET Local transactions](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/local-transactions) or [TransactionScopes](https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscope).
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:27]:
```suggestion
The outbox supports transactions via [ADO.NET local transactions](https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/local-transactions) or [TransactionScopes](https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscope).
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 3]
```diff
@@ -24,7 +24,28 @@ The trade-off is that each message processing attempt requires additional round
...
+
+### ADO.net
+
+By default the outbox uses the ADO.NET local transactions abstracted via `DbTransaction`. This is appropriate for most situations.
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:31]:
```suggestion
By default, the outbox uses the ADO.NET local transactions abstracted via `DbTransaction`. This is appropriate for most situations.
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 4]
```diff
@@ -24,7 +24,28 @@ The trade-off is that each message processing attempt requires additional round
...
+
+#### Isolation Level
+
+If required, the transaction [isolation level](https://learn.microsoft.com/en-us/dotnet/api/system.data.isolationlevel) may be adjusted:
+
+snippet: SqlPersistenceOutboxIsolationLevel
+
+A change in the isolation level affects all data access included in transactions. This should be done only if the business logic executed by message handlers within outbox transactions requires different than the default (`ReadCommited`) isolation level to guarantee correctness.
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:39]:
```suggestion
A change in the isolation level affects all data access included in transactions. This should be used only if the business logic executed by message handlers within outbox transactions requires a different isolation level than the default (`ReadCommited`) to guarantee correctness.
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 5]
```diff
@@ -33,20 +54,27 @@ In cases where the outbox transaction spans multiple databases, the `Transaction
 snippet: SqlPersistenceOutboxTransactionScopeMode
 
 In this mode the SQL persistence creates a `TransactionScope` that wraps the whole message processing attempt and within that scope it opens a connection, that is used for:
- - storing the outbox record
- - persisting the application state change applied via `SynchronizedStorageSession`
 
-In addition to that connection managed by NServiceBus, users can open their own database connections in the message handlers. If the underlying database technology supports distributed transactions managed by the Microsoft Distributed Transaction Coordinator (MSDTC) (e.g. SQL Server, Oracle or PostgreSQL), the transaction is escalated to a distributed transaction.
+- storing the outbox record
+- persisting the application state change applied via `SynchronizedStorageSession`
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:59]:
```suggestion
- persisting the application state changes applied via `SynchronizedStorageSession`
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 6]
```diff
@@ -33,20 +54,27 @@ In cases where the outbox transaction spans multiple databases, the `Transaction
...
+
+In addition to that connection managed by NServiceBus, users can open their own database connections in the message handlers. If the underlying database technology supports distributed transactions managed by the Microsoft Distributed Transaction Coordinator (MSDTC) (e.g. SQL Server or Oracle), the transaction is escalated to a distributed transaction.
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:61]:
```suggestion
In addition to the connection managed by NServiceBus, users can open their own database connections in message handlers. If the underlying database technology supports distributed transactions managed by the Microsoft Distributed Transaction Coordinator (MSDTC) (e.g. SQL Server or Oracle), the transaction is escalated to a distributed transaction.
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 7]
```diff
@@ -33,20 +54,27 @@ In cases where the outbox transaction spans multiple databases, the `Transaction
...
+#### Isolation Level
 
-If required, the outbox transaction isolation level may be adjusted:
+If required, the transaction [isolation level](https://learn.microsoft.com/en-us/dotnet/api/system.transactions.isolationlevel?) may be adjusted:
 
-snippet: SqlPersistenceOutboxIsolationLevel
+snippet: SqlPersistenceOutboxTransactionScopeMode
 
-A change in the isolation level affects all data access included in transactions. This should be done only if the business logic executed by message handlers within outbox transactions requires higher than the default isolation level (`Read Committed`) to guarantee correctness (e.g. `Repeatable Read` or `Serializable`). The isolation level may also be adjusted when not using `TransactionScope` mode.
+A change in the isolation level affects all data access included in transactions. This should be done only if the business logic executed by message handlers within outbox transactions requires lower than the default isolation level (`Serializable`) to guarantee correctness.
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:71]:
```suggestion
A change in the isolation level affects all data access included in transactions. This should be done only if the business logic executed by message handlers within outbox transactions requires a lower isolation level than the default (`Serializable`) to guarantee correctness.
```

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md [hunk 8]
```diff
@@ -33,20 +54,27 @@ In cases where the outbox transaction spans multiple databases, the `Transaction
...
 The `TransactionScope` mode is most useful in legacy scenarios such as when migrating from the MSMQ transport to a messaging infrastructure that does not support MSDTC. In this scenario, it is no longer possible to use a distributed transaction which includes the transport and the database. To maintain consistency, the outbox must be used instead. If the outbox table cannot be added to the legacy database, it may be placed in a separate database, but access to both databases must be included in distributed transactions.
 
-#### Transaction Isolation Level
+#### Isolation Level
 
-If required, the outbox transaction isolation level may be adjusted:
+If required, the transaction [isolation level](https://learn.microsoft.com/en-us/dotnet/api/system.transactions.isolationlevel?) may be adjusted:
 
-snippet: SqlPersistenceOutboxIsolationLevel
+snippet: SqlPersistenceOutboxTransactionScopeMode
```
Comment (lailabougria) [inline persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:69]: If the snippet name changed why doesn't it appear in the snippet code as a change?

**Comment by andreasohlund** [comment]
https://github.com/Particular/NServiceBus.Persistence.Sql/pull/1652/files adds tests to verify all modes mentioned in this PR

File: persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md (hunk 8 above)
Comment (andreasohlund) [reply persistence/sql/outbox_modes_sqlpersistence_[6,).partial.md:69]: It already existed

**Review by tmasternak — APPROVED** [review]

**Pull request merged**